assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -41,10 +47,57 @@ interface Store {
 	ref?: HTMLElement;
 }
 
+interface Event {
+	keyCode: number;
+}
+
 type SelectorsStore = Pick< Store, 'context' | 'selectors' | 'ref' >;
 
 interactivityApiStore( {
 	state: {},
+	effects: {
+		woocommerce: {
+			keyboardAccess: ( store: Store ) => {
+				const { context, actions } = store;
+				let allowNavigation = true;
+
+				const handleKeyEvents = ( event: Event ) => {
+					if (
+						! allowNavigation ||
+						! context.woocommerce?.isDialogOpen
+					) {
+						return;
+					}
+
+					// Disable navigation for a brief period to prevent spamming.
+					allowNavigation = false;
+
+					requestAnimationFrame( () => {
+						allowNavigation = true;
+					} );
+
+					// Check if the esc key is pressed.
+					if ( event.keyCode === 27 ) {
```

Comment:
I'm wondering if we should create an Enum or a Constant for these key codes. One reason for this is that someone might edit this code and forget to update the comment, resulting in outdated comment.